libqtile/backend/wayland/drawer.py

```diff
@@ -50,13 +56,23 @@ def draw(
         if offsetx > self._win.width:  # type: ignore
             return
 
-        # Only attempt to run RecordingSurface's operations if it actually has
-        # some. self.surface.ink_extents() computes the bounds of the current
-        # list of operations. If any of the bounds are not 0.0 then the surface
-        # has operations and we should paint them.
-        if not any(not math.isclose(0.0, i) for i in self.surface.ink_extents()):
+        # We can't test for the surface's ink_extents here on its own as a completely
+        # transparent background would not show any extents but we may still need to
+        # redraw (e.g. if a Spacer widget has changed position and/or size)
+
+        # Check if current rect has changed since previous draw
+        current_rect = (offsetx, offsety, width, height)
+        rect_changed = current_rect != self.previous_rect
+
+        # Check if draw has content (would be False for completely transparent drawer)
+        ink_changed = any(not math.isclose(0.0, i) for i in self.surface.ink_extents())
```

Comment:
It's interesting that `ink_extents` doesn't take into account transparent painted things. I'm guessing you checked this before doing it this way? I'd have expected them to be handled in the same way as things without 100% transparency.

Reply:
Correct. Anything with 100% transparency doesn't register with `ink_extents()`. However, I think that makes sense as, if it's fully transparent, there's nothing to draw.